src/Collector/ClassNameCollector.php

```diff
@@ -33,9 +22,18 @@ public function satisfy(
         AstParserInterface $astParser
     ) {
         return preg_match(
-            '/'.$this->getRegexByConfiguration($configuration).'/i',
+            $this->getConfigurationRegex($configuration),
             $abstractClassReference->getClassName(),
             $collectorFactory
         );
     }
+
+    private function getConfigurationRegex(array $configuration)
```

Comment:
not sure what this `ByConfiguration` was supposed to mean, and I find a bit weird to return a string that's not a regex but is called regex... Again nitpicking, glad to revert if you feel it's unnecessary